cpp/src/aztec/dsl/turbo_proofs/turbo_proofs.cpp

```diff
@@ -0,0 +1,277 @@
+    init_circuit(cs);
+}
+void init_circuit(acir_format::acir_format cs)
+{
+    constraint_system = std::make_shared<acir_format::acir_format>(cs);
+}
+
+uint32_t c_get_circuit_size(uint8_t const* constraint_system_buf)
+{
+    auto constraint_system = from_buffer<acir_format::acir_format>(constraint_system_buf);
+    auto crs_factory = std::make_unique<bonk::ReferenceStringFactory>();
+    auto composer = create_circuit(constraint_system, std::move(crs_factory));
+
+    auto prover = composer.create_prover();
+    auto circuit_size = prover.get_circuit_size();
+
+    return static_cast<uint32_t>(circuit_size);
+}
+
+uint32_t c_get_exact_circuit_size(uint8_t const* constraint_system_buf)
+{
+    auto constraint_system = from_buffer<acir_format::acir_format>(constraint_system_buf);
+    auto crs_factory = std::make_unique<bonk::ReferenceStringFactory>();
+    auto composer = create_circuit(constraint_system, std::move(crs_factory));
+
+    auto num_gates = composer.get_num_gates();
+    return static_cast<uint32_t>(num_gates);
+}
+
+void init_proving_key(std::unique_ptr<bonk::ReferenceStringFactory>&& crs_factory)
+{
+    auto composer = create_circuit(*constraint_system, std::move(crs_factory));
+    proving_key = composer.compute_proving_key();
+}
+
+void init_verification_key(std::unique_ptr<bonk::ReferenceStringFactory>&& crs_factory)
+{
+    if (!proving_key) {
+        std::abort();
+    }
+    // Patch the 'nothing' reference string fed to init_proving_key.
+    proving_key->reference_string = crs_factory->get_prover_crs(proving_key->circuit_size);
+
+    verification_key = Composer::compute_verification_key_base(proving_key, crs_factory->get_verifier_crs());
+}
+
+plonk::TurboProver new_prover(std::vector<fr> witness)
+{
+    Composer composer(proving_key, nullptr);
+    create_circuit_with_witness(composer, *constraint_system, witness);
+
+    info("composer gates: ", composer.get_num_gates());
+
+    plonk::TurboProver prover = composer.create_prover();
+
+    return prover;
+}
+
+bool verify_proof(plonk::proof const& proof)
+{
+    TurboVerifier verifier(verification_key, Composer::create_manifest(verification_key->num_public_inputs));
+
+    std::unique_ptr<KateCommitmentScheme<turbo_settings>> kate_commitment_scheme =
+        std::make_unique<KateCommitmentScheme<turbo_settings>>();
+    verifier.commitment_scheme = std::move(kate_commitment_scheme);
+
+    return verifier.verify_proof(proof);
+}
+
+size_t c_composer__new_proof(void* pippenger,
+                             uint8_t const* g2x,
+                             uint8_t const* constraint_system_buf,
+                             uint8_t const* witness_buf,
+                             uint8_t** proof_data_buf)
+{
+
+    auto constraint_system = from_buffer<acir_format::acir_format>(constraint_system_buf);
+
+    auto crs_factory = std::make_unique<PippengerReferenceStringFactory>(
+        reinterpret_cast<scalar_multiplication::Pippenger*>(pippenger), nullptr, g2x);
```

Comment:
@kevaundray said he wanted to pass a nullptr as pippenger_lagrange because noir doesn't want to use it and we want things to fail fast if there's a problem.